src/Microsoft.AspNetCore.Razor.Runtime/Runtime/TagHelpers/TagHelperExecutionContext.cs

```diff
@@ -93,15 +73,11 @@ internal TagHelperExecutionContext(string tagName, TagMode tagMode)
                 throw new ArgumentNullException(nameof(endTagHelperWritingScope));
             }
 
+            Reinitialize(tagName, tagMode, items, uniqueId, executeChildContentAsync);
```

Comment:
Call this after initializing `_tagHelpers` and just `.Clear()` that field in `Reinitialize()`.